vowpalwabbit/io_item.h

```diff
@@ -0,0 +1,51 @@
+#ifndef _IO_ITEM_
+#define _IO_ITEM_
+class IO_Item {
+
+  public:
+
+      std::string message;
+      int numCharsInit;
+
+      IO_Item(){
+          numCharsInit = 0;
+      }
+
+      IO_Item(std::string myMsg, int myNumCharsInit){
+
+          message.assign(myMsg);
+          numCharsInit = myNumCharsInit;
+      }
+
+      IO_Item operator=(const IO_Item &toCopy){
```

Comment:
You don't need to specify the copy constructor or assignment operator since the default behavior will do what you want.